storage/fcds/test/store.go

```diff
@@ -102,6 +102,92 @@ func RunAll(t *testing.T, newStoreFunc func(t *testing.T) (fcds.Storer, func()))
 	t.Run("iterator", func(t *testing.T) {
 		RunIterator(t, newStoreFunc)
 	})
+
+	t.Run("next shard", func(t *testing.T) {
+		runNextShard(t, newStoreFunc)
+	})
+}
+
+// RunNextShard runs the test scenario for NextShard selection
+func runNextShard(t *testing.T, newStoreFunc func(t *testing.T) (fcds.Storer, func())) {
```

Comment:
@janos @zelig I am kind of not inclined to have this test so much. i have a feeling that the test mainly checks that the random number generator will generate the correct `magic` for a given seed. WDYT?

Reply:
@acud I think that we should tests for this behaviour, this or in some other form.